CDP4Dal/Session.cs

```diff
@@ -326,6 +304,17 @@ public async Task Open()
 
             CDPMessageBus.Current.SendMessage(new SessionEvent(this, SessionStatus.BeginUpdate));
             await this.Assembler.Synchronize(dtoThings);
+
+            this.ActivePerson = this.Assembler.Cache.Select(x => x.Value)

```

Comment:
Curiously, just made me think a bit. Doesn't this LINQ completely make the Lazyness of the cache irrelevant? We do a select on all value objects, initializing the lazies all at once. I wonder if this might be an unnecessary performance hog here.

Just something to digest.

Reply:
point taken, but i see no other way to access the query the active person when the only search keys we have are the type and a property which is only valid for that type.

The best way to access items in the cache is described here https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/wiki/Cache 

`Lazy` is primarily used to ensure atomicity as described here:

`An operation acting on shared memory is atomic if it completes in a single step relative to other threads. Despite being thread-safe, the code executed by delegates of AddOrUpdate and GetOrAdd methods of the ConcurrentDictionary is not subject to the atomicity of the operation. To solve that issue we use Lazy<T> types as Value of the ConcurrentDictionary. Although in this case we are using Lazy<T> mainly to ensure operation atomicity while working with the ConcurrentDictionary, deferring the creation of an object until it is first used also improves performance, avoids wasteful computation, and reduces program memory requirements.` 

If you can suggest a better query I am all ears